tests/test_parse_to_object.py

```
@@ -0,0 +1,409 @@
+        bit_str.push(b'\x9A\xBC', 16)          # 16 bits of data
+        result = BitstringParser().parse(bit_str)
+        res = self._to_dict(result)
+        # First 29 bits of 0x123456789ABC = 0x02468ACF
+        # Remaining 19 bits = 0x9ABC -> message data should be 0x009abc when hexified
+        assert res['msg_type'].startswith('0x')
+        # convert to int for comparison to avoid issues with leading zeros in hex strings
+        assert int(res['data']['msg_data'], 16) == int(b'\x00\x9a\xbc'.hex(), 16)
+        
+    def test_parse_bitstring_empty(self):
+        bit_str = BitString() #just an empty bitstring
+        with pytest.raises(IndexError) as e:
```

Ruff (0.14.8)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)